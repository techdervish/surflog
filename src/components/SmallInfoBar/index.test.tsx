import {
  createShallowWrapper,
  expectWrappersTreeToMatchSnapshot,
  mocks,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('SmallInfoBar component', () => {
  const mockFn = jest.fn();
  const wrapper = createWrapper({
    totalResultsFound: 5,
    onClickManage: mockFn,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('renders total results', () => {
    expect(wrapper.find('.__t').text()).toEqual('Found: 5');
  });

  it('calls onSearch on Dropdown change', () => {
    wrapper.find('a').simulate('click');
    expect(mockFn).toBeCalled();
  });
});
