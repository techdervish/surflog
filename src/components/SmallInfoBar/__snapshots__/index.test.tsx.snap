// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallInfoBar component renders the expected snapshot 1`] = `
<div
  className="flex justify-between fw2"
>
  <div
    className="pb2 f7 pt2 __t"
    style={
      Object {
        "marginLeft": -16,
      }
    }
  >
    Found: 5
  </div>
  <a
    className="db pv2 pointer fw5"
    onClick={[MockFunction]}
    style={
      Object {
        "color": "#373232",
        "marginRight": -16,
        "textDecoration": "underline",
        "textIndent": 6,
      }
    }
  >
    Manage records
  </a>
</div>
`;
